### PR TITLE
Run test with verbose mode and log to a file

### PR DESCRIPTION
**Pull request by preetapan**
This makes nomad tests more similar to Consul. With verbose mode you get more detail on failed tests, especially if they also have log statements. All test output is logged to a file and failed tests are highlighted at the end of the run, making it easy to see what failed.

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -228,10 +229,15 @@ test-nomad: dev ## Run Nomad test suites
...
 		go test \
+			-v \
 			-cover \
 			-timeout=900s \
-			-tags="nomad_test $(if $(HAS_LXC),lxc)" \
-			./...
+			-tags="nomad_test $(if $(HAS_LXC),lxc)" ./... >test.log ; echo $$? > exit-code
+	@echo "Exit code: $$(cat exit-code)" >> test.log
+	@grep -A1 -- '--- FAIL:' test.log || true
+	@grep '^FAIL' test.log || true
```
Comment (chelseakomlo) [inline GNUmakefile]: Does this test file persist after the task has finished? If so, adding it to the .gitignore would be good.

Comment (preetapan) [reply GNUmakefile]: *.log is already ignored https://github.com/hashicorp/nomad/blob/master/.gitignore#L5

Comment (dadgar) [reply GNUmakefile]: Would searching for a panic and `# github.com` when there are build errors be useful?

Comment (preetapan) [reply GNUmakefile]: Panic is a good one to grep out, I will add that. Test build errors are already caught by the `^FAIL` grep above, and the test output will show something like `FAIL	github.com/hashicorp/nomad/nomad/deploymentwatcher [build failed]
`.

**Comment by dadgar** [comment]
Cool LGTM once we add the panic

**Pull request merged**